kernel/GBEngine/kstd2.cc

```diff
@@ -3563,6 +3572,21 @@ poly kNF2 (ideal F,ideal Q,poly q,kStrategy strat, int lazyReduce)
   assume(strat->R==NULL);//omfree(strat->R);
   omfree(strat->S_2_R);
   omfree(strat->fromQ);
+#ifdef HAVE_SHIFTBBA
+  // only LM of elements in S is shifted
+  // necessary to prevent deleting the tail multiple times
+  if (strat->tailRing->isLPring && strat->Shdl != NULL && strat->Shdl->m != NULL)
+  {
+    for (int j = 0; j < strat->Shdl->nrows * strat->Shdl->ncols; j++)
```

Comment:
for an ideal:
for(int j=0;i<IDELEMS(strat->Shdl);j++)

Reply:
Will fix, but the same syntax is used in simpleideals.cc
```cpp
/// deletes an ideal/module/matrix
void id_Delete (ideal * h, ring r)
{
  if (*h == NULL)
    return;

  id_Test(*h, r);

  const int elems = (*h)->nrows * (*h)->ncols;

  if ( elems > 0 )
  {
    assume( (*h)->m != NULL );

    if (r!=NULL)
    {
      int j = elems;
      do
      {
        j--;
        poly pp=((*h)->m[j]);
        if (pp!=NULL) p_Delete(&pp, r);
      }
      while (j>0);
    }

    omFreeSize((ADDRESS)((*h)->m),sizeof(poly)*elems);
  }

  omFreeBin((ADDRESS)*h, sip_sideal_bin);
  *h=NULL;
}
```